src/main/kotlin/io/github/intellij/dlanguage/template/DlangContext.kt

```diff
@@ -0,0 +1,13 @@
+package io.github.intellij.dlanguage.template
+
+import com.intellij.codeInsight.template.TemplateContextType
+import com.intellij.psi.PsiFile
+
+open class DlangContext protected constructor() : TemplateContextType("D", "D Language") {
+
+    override fun isInContext(file: PsiFile, offset: Int): Boolean {
+        // Todo: in never IDE's (2018.2.x) the suffix check does not work. does the baseLanguage check superseeds this?
```

Comment:
Whats likely going on is that the psi file name is returning the name of the module, because the underlying psi node is a D module, afaik this has been the case for a while now, so I'm not quite sure why that check would fail only on older IDEs.  If you tried something like `file.virtualFile.name` you should get the name of the underlying file.